Objects/namespaceobject.c

```diff
@@ -124,9 +124,9 @@ namespace_repr(PyObject *ns)
         if (PyUnicode_Check(key) && PyUnicode_GET_LENGTH(key) > 0) {
             PyObject *value, *item;
 
-            value = PyDict_GetItemWithError(d, key);
-            if (value != NULL) {
+            if (PyDict_GetItemRef(d, key, &value) == 1) {
```

Comment:
If PyDict_GetItemRef returned 0 (indicating that the key was not present), won't this return NULL without an exception set? This can happen if the SimpleNamespace was mutated during the repr() call. That's definitely possible under free-threading; it might be possible in the default build if the key is a str subclass with an evil `__hash__` though I haven't been able to come up with a repro.

Reply:
Yeah, there are 3 cases:

* return value `-1` with an exception set: error
* return value `0` (value is `NULL`): not found
* return value `1` (value is not `NULL`): found

You should store the return value and only set *loop_error* to 1 if the return value is negative.